framework/src/problems/FEProblemBase.C

```diff
@@ -2955,6 +2971,18 @@ FEProblemBase::computeUserObjects(const ExecFlagType & type, const Moose::AuxGro
   // Finalize, threadJoin, and update PP values of Nodal
   finalizeUserObjects<NodalUserObject>(nodal);
 
+  // Initialize threaded GeneralUserObjects
+  initializeUserObjects<GeneralUserObject>(threaded_general);
+
+  if (threaded_general.hasActiveObjects())
+  {
+    ComputeThreadedGeneralUserObjectsThread ctguot(*this, threaded_general);
+    Threads::parallel_reduce(EmptyRange(), ctguot);
```

Comment:
This is what I was saying we shouldn't do.  You need to actually pass the range of userobjects here to ensure they all get executed.